Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -93,3 +93,72 @@ public extension Message {
   }
 }
 
+
+public extension Array where Iterator.Element: Message {
```

Comment:
> ... a facility to initialize message from JSONSerialization-decoded data. ...

Just as we have decoder methods to initialize a message from a JSON String or from JSON UTF8 data, we could initialize a message from the output of `JSONSerialization`.

That would allow people to do things like:
```
    let decodedJSON = JSONSerialization.jsonObject(with: data)
    let decodedJSONArray = ... verify it's an array ...
    let messageArray = try decodedJSONArray.map { try MyMessage(jsonObject: $0) }
```

This approach has an obvious disadvantage:  It would require adding what is essentially an entire new decoder and encoder to SwiftProtobuf and it would be slower than our current JSON support.

But it also has the advantage that people could use `JSONSerialization` to tear apart essentially arbitrary JSON (not just arrays) and then leverage SwiftProtobuf to type-check and bind pieces of that into generated messages.

To be clear:  This is just brainstorming right now.  I'm not actually proposing we do this.